qiskit/synthesis/evolution/qdrift.py

```diff
@@ -125,4 +132,8 @@ def expand(self, evolution: PauliEvolutionGate) -> list[tuple[str, tuple[int], f
         sampled_paulis = [
             (pauli[0], pauli[1], np.real(np.sign(pauli[2])) * rescaled_time) for pauli in sampled
         ]
+
+        if self.preserve_order:
```

Comment:
Only re-order if the order is _not_ preserved 🙂 
```suggestion
        if not self.preserve_order:
```